apps/api/v2/src/modules/workflows/workflows.repository.ts

```diff
@@ -0,0 +1,305 @@
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { PrismaWriteService } from "@/modules/prisma/prisma-write.service";
+import { UserWithProfile } from "@/modules/users/users.repository";
+import { TeamsVerifiedResourcesRepository } from "@/modules/verified-resources/teams-verified-resources.repository";
+import {
+  WorkflowActivationDto,
+  WorkflowTriggerDto,
+  WorkflowTriggerType,
+  WorkflowTimeUnit,
+  WorkflowStepDto,
+  WorkflowMessageDto,
+  RecipientType,
+  StepAction,
+  TemplateType,
+  CreateWorkflowDto,
+} from "@/modules/workflows/inputs/create-workflow.input";
+import { WorkflowOutput } from "@/modules/workflows/outputs/workflow.output";
+import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
+import { TimeUnit, Workflow, WorkflowStep, WorkflowTriggerEvents } from "@prisma/client";
+
+import { TUpdateInputSchema } from "@calcom/platform-libraries/workflows";
+import { updateWorkflow } from "@calcom/platform-libraries/workflows";
+import { PrismaClient } from "@calcom/prisma";
+
+export type WorkflowType = Workflow & { activeOn: { eventTypeId: number }[]; steps: WorkflowStep[] };
+
+@Injectable()
+export class WorkflowsRepository {
+  constructor(
+    private readonly dbRead: PrismaReadService,
+    private readonly dbWrite: PrismaWriteService,
+    private readonly teamsVerifiedResourcesRepository: TeamsVerifiedResourcesRepository
+  ) {}
+
+  async deleteTeamWorkflowById(teamId: number, workflowId: number) {
+    return await this.dbWrite.prisma.workflow.delete({ where: { id: workflowId, teamId } });
+  }
+
+  async getTeamWorkflowById(teamId: number, id: number): Promise<WorkflowType> {
+    const workflow = await this.dbRead.prisma.workflow.findUnique({
+      where: {
+        id: id,
+        teamId: teamId,
+      },
+      include: {
+        steps: true,
+        activeOn: { select: { eventTypeId: true } },
+      },
+    });
+
+    if (!workflow) {
```

Comment:
I don't think that an exception should be thrown on repository level - should just return null like our other repos and then service checks how to respond.

Same for other methods.